### PR TITLE
added application insights telemetry package to send metrics and logs

### PR DESCRIPTION
**Pull request by tamilmani1989**
… to appinsights

**What this PR does / why we need it**:
added application insights telemetry package to send metrics and logs. Application can use this package to push data and metrics to application insights and then to geneva

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by tamilmani1989** [comment]
One thing i noticed to acquire lock before writing metadata to json file..will do that

**Comment by tamilmani1989** [comment]
> One thing i noticed to acquire lock before writing metadata to json file..will do that

done

File: aitelemetry/metadata_test.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"location":"eastus","name":"k8s-agentpool1-42685608-0","offer":"aks","osType":"Linux","placementGroupId":"","platformFaultDomain":"0","platformUpdateDomain":"0","publisher":"microsoft-aks","resourceGroupName":"rgcnideftesttamil","sku":"aks-ubuntu-1604-201902","subscriptionId":"ea821859-912a-4d20-a4dd-e18a3ce5ba2c","tags":"aksEngineVersion:canary;creationSource:aksengine-k8s-agentpool1-42685608-0;orchestrator:Kubernetes:1.10.13;poolName:agentpool1;resourceNameSuffix:42685608","version":"2019.02.12","vmId":"6baf785b-397c-4967-9f75-cdb3d0df66c4","vmSize":"Standard_DS2_v2","KernelVersion":""}
```
[the remark this answers is not in the document]
Comment (tamilmani1989) [reply aitelemetry/metadata_test.json]: need this for unit test

File: common/utils.go [hunk 2]
```diff
@@ -158,3 +192,68 @@ func ReadFileByLines(filename string) ([]string, error) {
 
 	return lineStrArr, nil
 }
+
+// GetHostMetadata - retrieve VM metadata from wireserver
+func GetHostMetadata(fileName string) (Metadata, error) {
```
[the remark this answers is not in the document]
Comment (tamilmani1989) [reply common/utils.go]: this should be fine..why pointer?

File: aitelemetry/telemetrywrapper_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+)
+
+var th *TelemetryHandle
+
+func TestMain(m *testing.M) {
+
+	if runtime.GOOS == "linux" {
+		platform.ExecuteCommand("cp metadata_test.json /tmp/azuremetadata.json")
+	} else {
+		platform.ExecuteCommand("copy metadata_test.json azuremetadata.json")
```
[the remark this answers is not in the document]
Comment (tamilmani1989) [reply aitelemetry/telemetrywrapper_test.go]: this is fine.. its just unit test..

File: aitelemetry/telemetrywrapper_windows.go [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package aitelemetry
+
+const (
+	metadataFile = "azuremetadata.json"
```
[the remark this answers is not in the document]
Comment (tamilmani1989) [reply aitelemetry/telemetrywrapper_windows.go]: done

File: common/utils.go [hunk 5]
```diff
@@ -36,6 +46,30 @@ type XmlDocument struct {
...
+	SubscriptionID       string `json:"subscriptionId"`
+	Tags                 string `json:"tags"`
+	OSVersion            string `json:"version"`
+	VMID                 string `json:"vmId"`
+	VMSize               string `json:"vmSize"`
+	KernelVersion        string
+}
+
+type metadataWrapper struct {
+	Metadata Metadata `json:"compute"`
```
Comment (chandanAggarwal) [inline common/utils.go]: >compute [](start = 26, length = 7)

what's this?

File: aitelemetry/telemetrywrapper.go [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+		return nil
+	})
+}
+
+func getMetadata(th *telemetryHandle) {
+	var metadata common.Metadata
+	var err error
+
+	// check if metadata in memory otherwise initiate wireserver request
+	for true {
```
Comment (jaer-tsun) [inline aitelemetry/telemetrywrapper.go]: remove true

File: common/utils.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaer-tsun) [reply common/utils.go]: so you don't need to return empty Metadata

[the remark this answers is not in the document]
Comment (tamilmani1989) [reply common/utils.go]: since pass by value is better in golang..will stick with this

**Pull request merged**